op-program/client/l1/blob_fetcher.go

```diff
@@ -0,0 +1,37 @@
+package l1
+
+import (
+	"context"
+	"errors"
+
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+var InvalidHashesLengthError = errors.New("invalid hashes length")
+
+type BlobFetcher struct {
+	logger log.Logger
+	oracle Oracle
+}
+
+var _ = (*derive.L1BlobsFetcher)(nil)
+
+func NewBlobFetcher(logger log.Logger, oracle Oracle) *BlobFetcher {
+	return &BlobFetcher{
+		logger: logger,
+		oracle: oracle,
+	}
+}
+
+// GetBlobs fetches blobs that were confirmed in the given L1 block with the given indexed hashes. It does so by fetching
+// the 4096 field elements for each blob through the preimage oracle.
```

Comment:
nit remove the `4096` details, that is introspection into the blob reconstruction.
```suggestion
// GetBlobs fetches blobs confirmed in the given L1 block with the specified indexed hashes.
```